libvcx/src/api/vcx.rs

```diff
@@ -37,6 +155,62 @@ pub extern fn vcx_init_core(config: *const c_char) -> u32 {
     }
 }
 
+/// Opens pool based on vcx configuration passed as a parameter
+///
+/// #Params
+/// command_handle: command handle to map callback to user context.
+///
+/// pool_config: Pool configuration
+/// {
+///     "genesis_path" - Path to the genesis file
+///     "pool_name" (optional) - Name of the pool ledger configuration.
+///     `pool_config` (optional) - Runtime pool configuration json as a string.
+///                         if NULL, then default config will be used.
+/// }
+/// where pool config structure is as follows
+/// {
+///     "timeout": int (optional), timeout for network request (in sec).
+///     "extended_timeout": int (optional), extended timeout for network request (in sec).
+///     "preordered_nodes": array<string> -  (optional), names of nodes which will have a priority during request sending:
+///         ["name_of_1st_prior_node",  "name_of_2nd_prior_node", .... ]
+///         This can be useful if a user prefers querying specific nodes.
+///         Assume that `Node1` and `Node2` nodes reply faster.
+///         If you pass them Libindy always sends a read request to these nodes first and only then (if not enough) to others.
+///         Note: Nodes not specified will be placed randomly.
+///     "number_read_nodes": int (optional) - the number of nodes to send read requests (2 by default)
+///         By default Libindy sends a read requests to 2 nodes in the pool.
+///         If response isn't received or `state proof` is invalid Libindy sends the request again but to 2 (`number_read_nodes`) * 2 = 4 nodes and so far until completion.
+/// }
+///
+/// cb: Callback that provides error status
+///
+/// #Returns
+/// Error code as a u32
+#[no_mangle]
+pub extern fn vcx_open_pool_directly(command_handle: CommandHandle, pool_config: *const c_char,  cb: extern fn(xcommand_handle: CommandHandle, err: u32)) -> u32 {
```

Comment:
Maybe `vcx_open_main_pool` for symmetry? It's the same case, right now we have 1 main pool opened, but perhaps in the future 1 agent can be connected to multiple ledgers.